### ISSUE TITLE
Replace Pojo with unknown?

### ISSUE THREAD

**Issue by yfwz100** [tskj/typescript-json-decoder#15]
Pojo is a library-defined type which generate an unnecessary dependency from 'typescript-json-decoder'. Will you consider using `unknown` instead of Pojo as the input of the decoder function.

**Comment by tskj** [comment]
Interesting observation, that's probably true. I can't think of what consequences this might have at the top of my head, but I will check it out when I am back to my computer over the weekend.

Do you mind giving an example of your usecase to give me a clearer understanding?

**Comment by yfwz100** [comment]
For 2 scenarios:

1. When I want to use this lib to decode the data from `node-fetch`... Newer version of `node-fetch` returns the `unknown` type when getting the JSON:
   ```js
   import fetch from 'node-fetch';

   const resp = await fetch('...');
   const data = await resp.json(); // now data is unknown.

   const result = decode(data); // where decode is define via `typescript-json-decoder`, it reports error because decode needs a Pojo type as input instead of unknown.
   ```

2. When I want to decouple modules... If I want to make an abstract validation function, I still need to `import { Pojo } from 'typescript-json-decoder';` to get the definition of Pojo.

**Comment by tskj** [comment]
That seems super reasonable! Thanks for the suggestion, I have now fixed this. 

I have released a new version (1.0.9) that changes all decoders to accept `unknown` and then do a runtime assert to verify that it's a POJO-like object that can be parsed. The `Pojo` type is still exported for backwards compatibility or if you'd just like to use it, but it shouldn't be required anymore.

**Comment by yfwz100** [comment]
Thanks for the changes! It helps a lot.
